### PR TITLE
Add line reader

### PR DESCRIPTION
**Pull request by poroshinleonid**
The line reader shows the prompt to the user and waits for a shell command.
Reader stores history automatically.

### CODE REVIEW

File: include/reader.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/* ************************************************************************** */
+
+#ifndef READER_H
+# define READER_H
+
+# include <stdio.h>
+# include <stdlib.h>
+# include <readline/readline.h>
+# include <readline/history.h>
+
```
Comment (taaae) [inline include/reader.h:20]: move includes to reader.c
src: https://google.github.io/styleguide/cppguide.html#Include_What_You_Use

File: src/reader/reader.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+char	*get_line(void)
+{
+	char			*line;
+
+	line = readline(SHELL_PROMPT);
+	if (line == NULL || line[0] == '\0')
+		return (NULL);
+	add_history(line);
+	return (line);
+}
```
Comment (taaae) [inline src/reader/reader.c:35]: how are we going to use history in the shell? if we gonna handle arrow-up to iterate through the history or 'history' linux command, we might need additional functions like getting a last command from the history.

**Review by taaae — CHANGES_REQUESTED** [review]
maybe we need to set up tests before merging anything

File: src/reader/reader.c (hunk 2 above)
Comment (poroshinleonid) [reply src/reader/reader.c:39]: we dont need to do it

File: include/reader.h (hunk 1 above)
Comment (poroshinleonid) [reply include/reader.h]: done

**Pull request merged**